fix #9193: hard-code point instead of language-specific separator

eddiemuc opened the pull request:
fix #9193: hard-code point instead of language-specific separator

UniQP commented:
@eddiemuc Have you seen https://github.com/cgeo/cgeo/issues/9193#issuecomment-710162730 ?

eddiemuc commented:
> @eddiemuc Have you seen [#9193 (comment)](https://github.com/cgeo/cgeo/issues/9193#issuecomment-710162730) ?

I don't understand this comment. Can you elaborate?

UniQP commented on main/src/cgeo/geocaching/location/GeopointFormatter.java:
```diff
@@ -98,8 +98,8 @@ public static String format(final Format format, final Geopoint gp) {
 
             case LAT_LON_DECMINUTE_SHORT:
                 return String.format(Locale.getDefault(), "%c%d %02d%c%03d %c%d %02d%c%03d",
```
```suggestion
                return String.format((Locale) null, "%c%d %02d.%03d %c%d %02d.%03d",
```
and remove the `'.'` arguments.

UniQP replied:
@eddiemuc Also applies to new version

eddiemuc replied:
done

UniQP commented on main/src/cgeo/geocaching/location/GeopointFormatter.java:
```diff
@@ -98,8 +98,8 @@ public static String format(final Format format, final Geopoint gp) {
 
             case LAT_LON_DECMINUTE_SHORT:
```
The example in the enum comment also needs to be adapted and should mention that the string in unlocalized.

UniQP reviewed (changes requested):
If we used the unlocalized version for DECMINUTE, we should also use it for other formats (DECSECOND, …).

eddiemuc commented:
> If we used the unlocalized version for DECMINUTE, we should also use it for other formats (DECSECOND, …).

Have you checked whether this has no unwanted side effects on other parts of c:geo?

eddiemuc commented:
HI @UniQP thanks for the review. You made good points, I decided to approach this a different way based on your feedback.

I now introduced a NEW enum value `LAT_LON_DECMINUTE_SHORT_POINTHARDCODED` and changed the relevant placed to use this one instead of the other value `LAT_LON_DECMINUTE_SHORT` when exporting waypoints to persnoal note. That way we can be sure I don't break existing functionality where maybe the language-specific settings are wanted and needed.

I squashed commits and force-pushed the change, maybe you (or someone) could take a second look?

UniQP commented on main/src/cgeo/geocaching/location/GeopointFormatter.java:
```diff
@@ -30,6 +30,9 @@ private GeopointFormatter() {
         /** Example: "N10 12,345 W5 12,345" */
         LAT_LON_DECMINUTE_SHORT,
 
+        /** Example: "N10 12.345 W5 12.345" (always use point, independent from language settings) */
```
```suggestion
        /** Example: "N10 12.345 W5 12.345" (unlocalized) */
```

eddiemuc replied:
done

UniQP commented on main/src/cgeo/geocaching/location/GeopointFormatter.java:
```diff
@@ -30,6 +30,9 @@ private GeopointFormatter() {
         LAT_LON_DECMINUTE_SHORT,
 
+        /** Example: "N10 12.345 W5 12.345" (always use point, independent from language settings) */
+        LAT_LON_DECMINUTE_SHORT_POINTHARDCODED,
```
```suggestion
        LAT_LON_DECMINUTE_SHORT_RAW,
```

eddiemuc replied:
done

eddiemuc commented:
@UniQP I believe I fixed all your comments according to your suggestions (I am a little overwhelmed by GitHub GUI, so I really hope I don't miss anything). If there's anything left, please drop a short hint.
